pkg/genericclioptions/genericclioptions.go

```diff
@@ -23,13 +28,120 @@ type IOStreams struct {
 	ErrOut io.Writer
 }
 
-type ClientConfig struct {
-	Kubeconfig    string
+type sharedClientConfig struct {
 	QPS           float32
 	Burst         int
 	UserAgentName string
-	RestConfig    *restclient.Config
-	ProtoConfig   *restclient.Config
+}
+
+func newSharedClientConfig(userAgentName string) sharedClientConfig {
+	return sharedClientConfig{
+		QPS:           50,
+		Burst:         75,
+		UserAgentName: userAgentName,
+	}
+}
+
+func (scc *sharedClientConfig) AddFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().Float32VarP(&scc.QPS, "qps", "", scc.QPS, "Maximum allowed number of queries per second.")
+	cmd.PersistentFlags().IntVarP(&scc.Burst, "burst", "", scc.Burst, "Allows extra queries to accumulate when a client is exceeding its rate.")
+}
+
+func (scc *sharedClientConfig) Validate() error {
+	return nil
+}
+
+func (scc *sharedClientConfig) Complete() error {
+	return nil
+}
+
+type ClientConfigSet struct {
+	sharedClientConfig
+	Kubeconfigs   []string
+	ClientConfigs []ClientConfig
+}
+
+func NewClientConfigSet(userAgentName string) ClientConfigSet {
+	return ClientConfigSet{
+		sharedClientConfig: newSharedClientConfig(userAgentName),
+		Kubeconfigs:        []string{defaultKubeconfig},
+	}
+}
+
+func (ccs *ClientConfigSet) AddFlags(cmd *cobra.Command) {
+	ccs.sharedClientConfig.AddFlags(cmd)
+
+	cmd.PersistentFlags().StringArrayVarP(&ccs.Kubeconfigs, "kubeconfig", "", ccs.Kubeconfigs, "Path to kubeconfig file(s).")
+}
+
+func (ccs *ClientConfigSet) Validate() error {
+	var errs []error
+	var err error
+
+	err = ccs.sharedClientConfig.Validate()
+	if err != nil {
+		errs = append(errs, fmt.Errorf("invalid shared client config: %w", err))
+	}
+
+	if len(ccs.Kubeconfigs) == 0 {
+		errs = append(errs, fmt.Errorf("invalid kubeconfigs: at least one kubeconfig must be provided"))
```

Comment:
Parameter is called `kubeconfig`
```suggestion
		errs = append(errs, fmt.Errorf("invalid kubeconfig: at least one kubeconfig must be provided"))
```

Reply:
done